blog/2019-09-ngrx-data-views/README.md

```diff
@@ -0,0 +1,419 @@
+
+There are multiple ways of tackling this issue:
+
+1. Cache the results of call expressions:
+
+   In this easy example you could just calculate the result once and keep it as a simple property on the `BookView` instance instead. Everything in the store will stay properly normalized, it is just the data view supplying derived data for easy access:
+
+   ```ts
+   export interface BookView {
+     ...
+     isNew: boolean;
+   }
+
+   export const getBookView = () =>
+     createSelector(
+       ...
+       (book, authors, tags): BookView =>
+         book && {
+           ...
+           isNew: calculateNew(book),
+         }
+     );
+   ```
+
+   In other cases the `ngOnChanges` in component classes could be used to cache expensive calculations to fields. And then those fields could be bound to on the templates.
+
+2. Disable Change Detection:
+
+   By choosing `OnPush` [change detection](https://angular.io/api/core/ChangeDetectionStrategy), the standard change detection triggered by every browser event can be switched off and change detection is only run when `@Input` decorated fields of a component are changing. This is a great way of optimizing Angular applications which use NgRx.
+
+3. Make repeated calls inexpensive:
+
+   For demonstration purposes we want to follow a different path. We want to memoize the result ourselves and by that assure that the expensive calculation is lazily evaluated and cached for further references.
+
+One way of implementing this is by using the `lodash` helper method [`once`](https://lodash.com/docs/#once), wrapping it around the method in the `getBookView` selector:
+
+```ts
+import { once } from 'lodash-es';
+
+export const getBookView = () =>
+  createSelector(
+    ...
+    (book, authors, tags): BookView =>
+      book && {
+        ...
+        isNew: once(() => calculateNew(book)),
+      }
+  );
+```
+
+Lodash imports should always be made by tree-shakeable libraries like [`lodash-es`](https://www.npmjs.com/package/lodash-es) or [`lodash.X`](https://www.npmjs.com/package/lodash.once) to reduce the bundle size. For the memoization of methods with parameters the helper [`memoize`](https://lodash.com/docs/#memoize) can be used in a similar fashion.
+
+The final result can be studied in the last [StackBlitz Example](https://stackblitz.com/github/dhhyi/ngrx-data-views/tree/memoized-business-objects?file=src%2Fapp%2Fstore%2Fbook-view%2Findex.ts&devtoolsheight=50).
+
+## Conclusion
+
+I hope you had fun reading this and that I found the right approach in displaying this rather complex topic. Always keep an open mind and let me remind you: if it is Open Source you are using, you can typically look behind the curtains and peek into the dirty details. It helps...
```

Comment:
Provide a real conclusion here not just a closing sentence ;-)
It should sum up what this article was about. 3-5 sentences are enough, but it's important to repeat to "burn it in".

Reply:
I tried, still sucks :angry:

Reply:
Actually, I like it by now :grin: